### PR TITLE
wallet: expose set/get walletcache attribute functionality in wallet …

### PR DESCRIPTION
**Pull request by ghost**
monero-project#5534
monero-project#5779

### CODE REVIEW

**Comment by tomato5040** [comment]
This is needed for new GUI

**Comment by ghost** [comment]
Would be good to have workflow access for some of these PR before merging @aeonix . Builds are good on ubuntu but I don't have chance to test other OS. Travis appears basically useless "docker: Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit."

File: src/wallet/api/wallet.cpp [hunk 1]
```diff
@@ -1608,7 +1608,9 @@ bool WalletImpl::setCacheAttribute(const std::string &key, const std::string &va
 
 std::string WalletImpl::getCacheAttribute(const std::string &key) const
 {
-    return m_wallet->get_attribute(key);
+    std::string value;
+    m_wallet->get_attribute(key, value);
+    return value;
```
Comment (stoffu) [inline src/wallet/api/wallet.cpp:1613]: Please add /monero#5779 to the end of the commit message.

Comment (ghost) [reply src/wallet/api/wallet.cpp:1613]: Resolved.

**Review by stoffu — APPROVED** [review]

**Comment by BigslimVdub** [comment]
> Would be good to have workflow access for some of these PR before merging @aeonix . Builds are good on ubuntu but I don't have chance to test other OS. Travis appears basically useless "docker: Error response from daemon: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit."

That’s due to github monetization and limiting build use per month unless you are on a paid plan. Too many builds and it blocks further builds for the month. 

If you any others to try builds on other OS you should reach out to community members as there are plenty that can help fetch PR and build on local OS, or you can create some vm’s as well.

**Pull request closed without merging**